packages/suite/src/hooks/settings/backends/useBackendsForm.ts

```diff
@@ -119,22 +120,28 @@ export const useBackendsForm = (coin: Network['symbol']) => {
 
     const input = useBackendUrlInput(coin, currentValues.type, currentValues.urls);
 
-    const save = () => {
-        const { type, urls } = currentValues;
+    const getUrls = () => {
         const lastUrl = input.value && !input.error ? [input.value] : [];
-        const defaultType = getDefaultBackendType(coin);
+        return currentValues.urls.concat(lastUrl);
+    };
 
+    const hasOnion = () => !!getUrls().find(isOnionUrl);
```

Comment:
the logic has to be changed:

- it's not important if there is some onion address but if there are only onion addresses

The only state that has to be handled with the modal dialog is:
User with only onion custom backends is trying to disable tor.